### PR TITLE
Refactor ToggleControl to use functional component

### PR DESCRIPTION
**Pull request by sc81**

### CODE REVIEW

File: packages/components/src/toggle-control/index.js [hunk 1]
```diff
@@ -7,62 +7,48 @@ import classnames from 'classnames';
...
-		this.onChange = this.onChange.bind( this );
+export default function ToggleControl( {
+	label,
+	checked,
+	help,
+	className,
+	onChange,
+} ) {
+	function onChangeToggle( event ) {
+		onChange( event.target.checked );
```
Comment (youknowriad) [inline packages/components/src/toggle-control/index.js:26]: The previous onChange handler was checking whether the onChange prop was set or not. Should we do the same?

**Review by youknowriad — APPROVED** [review]
Thanks, this looks good.

Left a minor comment.

File: packages/components/src/toggle-control/index.js (hunk 1 above)
Comment (sc81) [reply packages/components/src/toggle-control/index.js:26]: The onChange prop is required, without this prop the component will not be usable. When prop is undefined the JS engine will report an error, so the programmer will know what he did wrong.

Comment (youknowriad) [reply packages/components/src/toggle-control/index.js:26]: Makes sense and it's also documented as such in the README.

**Pull request merged**